server/lib/service/dv360.dart

```diff
@@ -31,11 +34,58 @@ class DisplayVideo360Client {
         updateMask: 'entityStatus');
   }
 
+  /// Duplicates the line item to [insertionOrderIdDestination].
+  ///
+  /// The [advertiserIdDestination] is the ID of the destination advertiser.
+  ///
+  /// Throws an [ApiRequestError] if API returns an error.
+  Future<void> duplicateLineItem(Int64 advertiserId, Int64 lineItemId,
+      Int64 advertiserIdDestination, Int64 insertionOrderIdDestination) async {
+    final source = await _api.advertisers.lineItems
+        .get(advertiserId.toString(), lineItemId.toString());
+
+    // Stores the current entity status because only ENTITY_STATUS_DRAFT is
+    // allowed at creation.
+    final currentStatus = source.entityStatus;
+
+    // Sets the correct fields for line item creation.
+    // Inherits flight dates from the destination parent insertion order.
+    source
+      ..insertionOrderId = insertionOrderIdDestination.toString()
+      ..entityStatus = 'ENTITY_STATUS_DRAFT'
+      ..advertiserId = null
+      ..campaignId = null
+      ..lineItemId = null
+      ..updateTime = null
+      ..partnerCosts = null
+      ..name = null
+      ..flight.dateRange = null
```

Comment:
why not reuse the flight from source line item?

Reply:
When I was writing it, I wasn't sure if the flight date should be inherited from the destination parent insertion order. When it does inherit, the dateRange is output only.

So we should make the new flightDateType LINE_ITEM_FLIGHT_DATE_TYPE_CUSTOM and have the same dateRange as the source?

Reply:
i see. actually although it is input only, you can still set it, the server just ignore the input, :p, the better way is to add check `If isInherited`, make dateRange = null.

Reply:
Ok gotcha!